### PR TITLE
Refactor Functions

### PR DESCRIPTION
**Pull request by calgray**
As a preliminary for merging experimental changes, this is a set of non funtional refactoring to reduce diff sizes.

### CODE REVIEW

**Review by rtobar — COMMENTED** [review]
Nice, I think I also saw `cuda::PhaseMatrixFunction` function go away, is that right? That's good, IIRC it wasn't being used as both compute implementation used the one under `cpu`.

The only comment is about the downloading of data into the `mwa` subdirectory that needs to be checked for existence first, otherwise it looks good!

File: testdata/install.sh [hunk 1]
```diff
@@ -21,11 +21,11 @@
 # MA 02111-1307  USA
 #
 
-wget "https://cloudstor.aarnet.edu.au/plus/s/YoYdODmk9iVS5Sq/download" -O ./1197638568-32.tar.gz
-tar -C ./ -xvf ./1197638568-32.tar.gz
+wget "https://cloudstor.aarnet.edu.au/plus/s/YoYdODmk9iVS5Sq/download" -O ./mwa/1197638568-32.tar.gz
```
[the remark this answers is not in the document]
Comment (calgray) [reply testdata/install.sh]: good spot, just updated

**Review by rtobar — APPROVED** [review]
Looks better now. You'll get now an error message printed on screen if the directory *already* exists, but the script will continue executing anyway, so it's a minor aesthetic detail.

**Comment by calgray** [comment]
I can't remember if CI can detect warnings in install scripts (preferably yes) so I've taken out the mkdir warning

**Pull request merged**